### ISSUE TITLE
[WC 9.5-Beta-1] The `woocommerce_reports_charts` filter stopped working in WooCommerce 9.5-Beta-1.

### ISSUE THREAD

**Issue by iamdharmesh** [woothemes/woocommerce#52910]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The `woocommerce_reports_charts` filter stopped working in WooCommerce 9.5-Beta-1. The codebase below works well in WooCommerce 9.4 but stopped working in WC 9.5.

```php
add_filter( 'woocommerce_reports_charts', function ( $reports ) {
	$reports['tickets'] = array(
		'title'  =>  __( 'Tickets', 'textdomain' ),
		'charts' => array(
			array(
				'title'       => __( 'Sales by Product', 'textdomain' ),
				'description' => '',
				'hide_title'  => true,
				'function'    => function () {
					return [];
				}
			),
		)
	);
	return $reports;
} );
```

Upon investigation, I found that this issue is caused by PR https://github.com/woocommerce/woocommerce/pull/51592, where the report is being unset if the `reports` key is not present in the array. https://github.com/woocommerce/woocommerce/blob/c9f9d7475388d1d066cf4300e618023b1e27435c/plugins/woocommerce/includes/admin/class-wc-admin-reports.php#L220-L224

The unsetting should occur after the backward compatibility code. Placing this backward compatibility code before the unsetting resolves the issue.
https://github.com/woocommerce/woocommerce/blob/c9f9d7475388d1d066cf4300e618023b1e27435c/plugins/woocommerce/includes/admin/class-wc-admin-reports.php#L225-L227

### Expected behavior

The reports extended with the `woocommerce_reports_charts` filter should function correctly in WooCommerce 9.5-Beta-1.

_**Note**: I understand that `woocommerce_reports_charts` is an old filter, and we have the `woocommerce_admin_reports` filter, which works well with WC 9.5. Therefore, it’s better to stop using the `woocommerce_reports_charts` filter and switch to the `woocommerce_admin_reports` filter. Additionally, legacy reports were deprecated long ago, so this isn’t a significant issue. Feel free to close this as it’s more of an **FYI**-type report._

### Actual behavior

The reports extended with the `woocommerce_reports_charts` filter is not working in WooCommerce 9.5-Beta-1.

### Steps to reproduce

1. Place the code provided above in the theme's `functions.php` file.  
2. Navigate to `WooCommerce` > `Reports` and verify that the "Tickets" tab is visible on the reports page.  
3. Upgrade to WC 9.5-Beta-1.  
4. Go to `WooCommerce` > `Reports` again and observe that the "Tickets" tab is no longer visible.

### WordPress Environment

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.5.0-beta.1
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.2
Log Directory Writable: ✔
WP Version: 6.7
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 8.0.30
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 8.7.1
(SecureTransport) LibreSSL/3.3.6

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 91
custom_css: 1
event_ticket: 154
event_ticket_email: 19
mailpoet_page: 1
nav_menu_item: 17
page: 36
post: 17
postman_sent_mail: 64
product: 135
product_variation: 75
revision: 132
shop_coupon: 4
shop_order: 1744
shop_order_placehold: 1
shop_order_refund: 50
shop_subscription: 110
wc_booking: 6
wc_membership_plan: 1
wc_order_status: 12
wc_user_membership: 1
wgc_subscription: 20
wp_font_face: 33
wp_font_family: 12
wp_global_styles: 5
wp_navigation: 2
wp_template_part: 3
wpforms: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (4) ###

Email Log: by Sudar – 2.4.9
Query Monitor: by John Blackbourn – 3.16.4
WooCommerce Box Office: by WooCommerce – 1.2.9
WooCommerce: by Automattic – 9.5.0-beta.1

### Inactive Plugins (38) ###

10up WP-CLI Vulnerability Scanner: by 10up – 1.2.0
AutomateWoo: by WooCommerce – 6.0.8 (update to version 6.1.1 is available)
Braintree for WooCommerce Payment Gateway: by WooCommerce – 3.2.1 (update to version 3.2.2 is available)
Debug Bar: by wordpressdotorg – 1.1.6
DHL Express Live Rates: by Octolize – 4.0.0
Distributor Helper: by  –
Flexible Shipping: by Octolize – 5.0.3
Mailchimp for WooCommerce: by Mailchimp – 4.2.1
Plugin Check (PCP): by WordPress Performance Team and Plugin Review Team – 1.0.2
Restricted Site Access: by Jake Goldman
10up
Oomph – 7.3.4

User Switching: by John Blackbourn – 1.8.0
WooCommerce.com Update Manager: by Automattic – 1.0.3
WooCommerce Blocks: by Automattic – 11.7.0
WooCommerce Currency Converter: by WooCommerce – 1.6.29 (update to version 2.2.4 is available)
WooCommerce Deposits: by WooCommerce – 2.3.2 (update to version 2.3.3 is available)
WooCommerce Elavon Converge EU Gateway: by Elavon – 1.16.0
WooCommerce EU VAT Number: by WooCommerce – 2.9.8 (update to version 2.9.9 is available)
WooCommerce EU VAT Number SOAP Fix.: by  –
WooCommerce Eway Payment Gateway: by WooCommerce – 3.6.0 (update to version 3.7.2 is available)
WooCommerce GoCardless Gateway: by WooCommerce – 2.8.1 (update to version 2.8.2 is available)
WooCommerce Order Barcodes: by WooCommerce – 1.7.3 (update to version 1.8.1 is available)
WooCommerce Order Status Manager: by SkyVerge – 1.13.4 (update to version 1.15.4 is available)
WooCommerce Payfast Gateway: by WooCommerce – 1.6.7 (update to version 1.6.8 is available)
WooCommerce Pre-Orders: by WooCommerce – 2.1.4 (update to version 2.1.5 is available)
WooCommerce Product CSV Import Suite: by WooCommerce – 1.10.73 (update to version 1.10.74 is available)
WooCommerce Product Vendors: by WooCommerce – 2.3.1
WooCommerce Purchase Order Payment Gateway: by WooCommerce – 1.4.9
WooCommerce Shipping & Tax: by WooCommerce – 2.8.3
WooCommerce SnapScan Gateway: by WooCommerce – 1.2.2
WooCommerce Square: by WooCommerce – 4.7.2 (update to version 4.8.3 is available)
WooCommerce Stripe Gateway: by WooCommerce – 8.7.0
WooCommerce Subscriptions: by WooCommerce – 6.5.0 (update to version 6.9.0 is available)
WooCommerce Subscriptions CSV Importer and Exporter: by Prospress Inc – 2.2.0
WooCommerce WooCommerce GoCardless Gateway Test plugin: by  –
WooCommerce Xero Integration: by WooCommerce – 1.9.0 (update to version 1.9.1 is available)
WooPayments: by WooCommerce – 8.0.1 (update to version 8.5.0 is available)
WP Crontrol: by John Blackbourn – 1.17.0
WP Rollback: by WP Rollback – 2.0.7

### Dropin Plugins () ###

db.php: Query Monitor Database Class (Drop-in)

### Must Use Plugins (1) ###

ngrok.php: by  –

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: booking (booking)
converge-subscription (converge-subscription)
converge-variable-subscription (converge-variable-subscription)
external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude_from_catalog (exclude_from_catalog)
exclude_from_search (exclude_from_search)
exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
instock (instock)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: ✔

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 3 MB

### WC Pages ###

Shop base: #2889 - /shop-2/
Cart: #151 - /cart-block/ -  Contains the woocommerce/cart block
Checkout: #149 - /checkout-block/ -  Contains the woocommerce/checkout block
My account: #2892 - /my-account-2/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.6.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
Theme type: Classic theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: product-data-views
experimental-blocks
coming-soon-newsletter-template
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
blueprint
reactify-classic-payments-settings
use-wp-horizon
add-to-cart-with-options-stepper-layout

Daily Cron: ✔ Next scheduled: 2024-11-18 18:11:14 +00:00
Options: ✔
Notes: 81
Onboarding: completed

### Action Scheduler ###

Complete: 373
Oldest: 2024-10-20 07:26:49 +0000
Newest: 2024-11-18 14:25:54 +0000

Failed: 705
Oldest: 2023-02-16 07:47:32 +0000
Newest: 2024-11-18 14:19:13 +0000

Pending: 2
Oldest: 2024-11-18 15:11:57 +0000
Newest: 2024-11-19 12:06:19 +0000


### Status report information ###

Generated at: 2024-11-18 14:43:40 +00:00
```
</details>

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by tomalec** [comment]
Thanks, @iamdharmesh, for finding it!

Indeed, it's something I missed in https://github.com/woocommerce/woocommerce/pull/51592 :( 
It's a pitty we didn't find it earlier, as it's too late to push it to 9.5, and given how long deprecated reports are and `woocommerce_reports_charts` it's not worth for code freeze exception at this point.

Given how long deprecated it is and now, that I broke it, maybe it's an opportunity to remove it in 9.6 rather than reviving it back.

**Comment by iamdharmesh** [comment]
Hi @tomalec,

Thanks for checking on this.

>given how long deprecated reports are and `woocommerce_reports_charts` it's not worth for code freeze exception at this point.
>
>Given how long deprecated it is and now, that I broke it, maybe it's an opportunity to remove it in 9.6 rather than reviving it back.

Yes, I completely agree. As mentioned in the issue description, this is an FYI-type report. Please proceed with the best approach instead of fixing it.

Thank you.

**Comment by tomalec** [comment]
The fix was released to 9.5 as CFE https://github.com/woocommerce/woocommerce/issues/53137#issuecomment-2517281300
